bundles/org.openhab.core.automation.module.script.rulesupport/src/main/java/org/openhab/core/automation/module/script/rulesupport/loader/AbstractScriptFileWatcher.java

```diff
@@ -353,18 +358,14 @@ private boolean createAndLoad(ScriptFileReference ref) {
         return false;
     }
 
-    private void initialImport() {
-        File directory = watchPath.toFile();
-
-        if (!directory.exists()) {
-            if (!directory.mkdirs()) {
-                logger.warn("Failed to create watched directory: {}", watchPath);
-            }
-        } else if (directory.isFile()) {
-            logger.warn("Trying to watch directory {}, however it is a file", watchPath);
+    protected void performInitialImport() {
```

Comment:
I don't think this should be a separate method. If a sub-class wants to handle the import one it's own, it's free to do so and nesting methods only adds complexity without a real benefit.

Reply:
The only reason I made it a separate method, is so that there was this check inside the constructor:
```
        if (currentStartLevel > StartLevelService.STARTLEVEL_MODEL) {
```
and I don't want the child class to need to know this implementation detail, and instead it should [just call `performInitialImport()`](https://github.com/openhab/openhab-addons/pull/14590/files#diff-0391e0939589717a418a22a3138a32b9fe3b36a31091bb5bfbb8cd5bd9938101R66).

I couldn't move it into `initialImport` because it's not always checked with this exact condition everywhere, just inside the constructor.

Reply:
I realised that the two checks are essentially the same, and I've combined the two functions into one and use the same check.

Reply:
Thanks, can you keep the original name?

Reply:
I've changed it to the original name.